tests/core/vm/test_computation.py

```diff
@@ -0,0 +1,184 @@
+import pytest
+
+from evm.vm import (
+    Computation,
+    Message,
+)
+
+from eth_utils import (
+    to_canonical_address,
+)
+
+from evm.exceptions import (
+    ValidationError,
+)
+
+CANONICAL_ADDRESS_1 = to_canonical_address("0x0f572e5295c57f15886f9b263e2f6d2d6c7b5ec6")
+CANONICAL_ADDRESS_2 = to_canonical_address("0xcd1722f3947def4cf144679da39c4c32bdc35681")
+
+
+@pytest.fixture
+def message():
+    message = Message(
+        origin=to_canonical_address(CANONICAL_ADDRESS_2),
+        to=CANONICAL_ADDRESS_1,
+        sender=to_canonical_address(CANONICAL_ADDRESS_2),
+        value=100,
+        data=b"0x",
+        code=b"0x",
+        gas=100,
+        gas_price=1,
+    )
+    return message
+
+
+@pytest.fixture
+def computation(message):
+    computation = Computation(
+        vm=1,
+        message=message
+    )
+    return computation
+
+
+@pytest.fixture
+def create_error():
+    def computation_error(computation):
+        computation.gas_meter.gas_remaining = 0
+        with computation:
+            computation.extend_memory(32, 64)
+        return computation
+    return computation_error
+
+
+def test_prepare_child_message(computation):
+    child_message = computation.prepare_child_message(
+        gas=100,
+        to=to_canonical_address(CANONICAL_ADDRESS_2),
+        value=200,
+        data=b"0x",
+        code=b"0x"
+    )
+    assert computation.msg.depth == 0
+    assert child_message.depth == 1
+    assert computation.msg.origin == child_message.origin
+    assert computation.msg.gas_price == child_message.gas_price
+    assert computation.msg.storage_address == child_message.sender
+
+
+def test_extend_memory(computation):
+    with pytest.raises(ValidationError):
+        # Start positions needs to be a uint256
+        computation.extend_memory("1", 1)
+        computation.extend_memory(-1, 1)
+        # Size needs to be a uint256
+        computation.extend_memory(1, "1")
+        computation.extend_memory(1, -1)
+    # Stays the same if size is 0
+    assert len(computation.memory.bytes) == 0
+    computation.extend_memory(1, 0)
+    assert len(computation.memory.bytes) == 0
+    # Memory extends by 32 bytes and costs 3 gas
+    assert computation.gas_meter.gas_remaining == 100
+    computation.extend_memory(0, 1)
+    assert len(computation.memory.bytes) == 32
+    assert computation.gas_meter.gas_remaining == 97
+    # Memory isn't addded until 32 bytes are used
+    computation.extend_memory(1, 1)
+    assert len(computation.memory.bytes) == 32
+    computation.extend_memory(2, 32)
+    assert len(computation.memory.bytes) == 64
+    assert computation.gas_meter.gas_remaining == 94
+
+
+def test_register_accounts_for_deletion(computation):
+    # Fails if address isn't canonical
+    with pytest.raises(ValidationError):
+        computation.register_account_for_deletion("0x0f572e5295c57f15886f9b263e2f6d2d6c7b5ec6")
```

Comment:
Rather than using `"0x0f572e5295c57f15886f9b263e2f6d2d6c7b5ec6"` here, it seems like this value should be assigned as `NORMALIZED_ADDRESS_1` so that it's clear that it is related to `CANONICAL_ADDRESS_1`.